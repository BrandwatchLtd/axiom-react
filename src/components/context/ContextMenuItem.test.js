import React from 'react';
import renderer from 'react-test-renderer';
import { ContextMenuItem } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <ContextMenuItem { ...props }>
      Lorem ipsum
    </ContextMenuItem>
  );
}

describe('ContextMenuItem', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
