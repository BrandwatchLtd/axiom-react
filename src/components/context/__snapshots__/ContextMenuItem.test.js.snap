// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuItem renders with defaultProps 1`] = `
<li
  className="ax-context-menu__list-item"
>
  <button
    className="ax-context-menu__item-single"
    disabled={undefined}
    onClick={undefined}
  >
    <div
      className="ax-context-menu__item-content"
    >
      Lorem ipsum
    </div>
  </button>
</li>
`;

exports[`ContextMenuItem renders with disabled 1`] = `
<li
  className="ax-context-menu__list-item"
>
  <button
    className="ax-context-menu__item-single"
    disabled={true}
    onClick={undefined}
  >
    <div
      className="ax-context-menu__item-content"
    >
      Lorem ipsum
    </div>
  </button>
</li>
`;

exports[`ContextMenuItem renders with multiSelect 1`] = `
<li
  className="ax-context-menu__list-item"
>
  <div
    className="ax-context-menu__item-multi"
    disabled={undefined}
  >
    <label
      className="ax-chedio-buttox__container ax-space--x2"
      onClick={undefined}
    >
      <input
        checked={undefined}
        className="ax-chedio-buttox ax-checkbox"
        disabled={undefined}
        name={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <span
        className="ax-chedio-buttox__indicator ax-checkbox__indicator"
      />
      <span
        className="ax-chedio-buttox__label"
      >
        Lorem ipsum
      </span>
    </label>
  </div>
</li>
`;

exports[`ContextMenuItem renders with multiSelect and selected 1`] = `
<li
  className="ax-context-menu__list-item ax-text--strong"
>
  <div
    className="ax-context-menu__item-multi"
    disabled={undefined}
  >
    <label
      className="ax-chedio-buttox__container ax-space--x2"
      onClick={undefined}
    >
      <input
        checked={true}
        className="ax-chedio-buttox ax-checkbox"
        disabled={undefined}
        name={undefined}
        onChange={undefined}
        type="checkbox"
      />
      <span
        className="ax-chedio-buttox__indicator ax-checkbox__indicator"
      />
      <span
        className="ax-chedio-buttox__label"
      >
        Lorem ipsum
      </span>
    </label>
  </div>
</li>
`;

exports[`ContextMenuItem renders with selected 1`] = `
<li
  className="ax-context-menu__list-item ax-text--strong"
>
  <button
    className="ax-context-menu__item-single ax-context-menu__item-single--selected"
    disabled={undefined}
    onClick={undefined}
  >
    <div
      className="ax-context-menu__item-content"
    >
      Lorem ipsum
    </div>
    <div
      className="ax-context-menu__item-icon"
    >
      <svg
        className="ax-icon ax-icon--tick"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M4 7.5l2 2 6-6L13.5 5 6 12.5 2.5 9z\\" fill-rule=\\"nonzero\\" fill=\\"currentColor\\"/>",
          }
        }
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
        viewBox="0 0 16 16"
      />
    </div>
  </button>
</li>
`;
